### PR TITLE
[Enhancement] Print hint message for waiting globalStateMgr to be ready

### PR DESCRIPTION
**Pull request by gengjun-git**
Why I'm doing:

What I'm doing:
Print hint message for waiting globalStateMgr to be ready.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 1]
```diff
@@ -1215,6 +1218,20 @@ public void waitForReady() throws InterruptedException {
 
             Thread.sleep(2000);
             LOG.info("wait globalStateMgr to be ready. FE type: {}. is ready: {}", feType, isReady.get());
+
+            if (System.currentTimeMillis() - waitStartTime > printHintTimeInterval * printHintCnt) {
+                printHintCnt++;
+                LOG.warn("It took too much time to get ready, there are some reasons: \n" +
+                        "1. There are too many BDB logs to replay, because the failure of checkpoint. \n" +
+                        "2. More than half of the followers in the cluster are not started. \n" +
+                        "3. There are multiple IPs on the machine, " +
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1227]: or ip has changed?

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 2]
```diff
@@ -1215,6 +1218,20 @@ public void waitForReady() throws InterruptedException {
...
+                        "you should specify the priority_networks to use the IP from image/ROLE, " +
+                        "ignore this reason if you are using FQDN. \n" +
+                        "4. The time deviation between machines is greater than 5s, " +
+                        "please use ntp or other tools to synchronize clock. \n" +
+                        "5. The configuration of edit_log_port is changed, please reset to the original value. \n" +
+                        "6. The thread of replayer is blocked, please use jstack to find the details");
+            }
         }
     }
```
[the remark this answers is not in the document]
Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1237]: better to use sinle line log

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java]: done

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1237]: done

**Review by nshangyiming — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 3]
```diff
@@ -1215,6 +1218,20 @@ public void waitForReady() throws InterruptedException {
...
+                LOG.warn("It took too much time to get ready, there are some reasons: " +
+                        "1. There are too many BDB logs to replay, because the failure of checkpoint. " +
+                        "2. More than half of the followers in the cluster are not started. " +
+                        "3. There are multiple IPs on the machine, or ip has changed, " +
+                        "you should specify the priority_networks to use the IP from image/ROLE, " +
+                        "ignore this reason if you are using FQDN. " +
+                        "4. The time deviation between machines is greater than 5s, " +
+                        "please use ntp or other tools to synchronize clock. " +
+                        "5. The configuration of edit_log_port is changed, please reset to the original value. " +
+                        "6. The thread of replayer is blocked, please use jstack to find the details");
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1233]:
```suggestion
                        "6. The replayer thread may get stuck, please use jstack to find the details.");
```

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 4]
```diff
@@ -1215,6 +1218,20 @@ public void waitForReady() throws InterruptedException {
...
+            }
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1234]: 7. The remaining capacity of disk where FE meta dir is on is under 5GB.

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 5]
```diff
@@ -1215,6 +1218,20 @@ public void waitForReady() throws InterruptedException {
 
             Thread.sleep(2000);
             LOG.info("wait globalStateMgr to be ready. FE type: {}. is ready: {}", feType, isReady.get());
+
+            if (System.currentTimeMillis() - waitStartTime > printHintTimeInterval * printHintCnt) {
+                printHintCnt++;
+                LOG.warn("It took too much time to get ready, there are some reasons: " +
+                        "1. There are too many BDB logs to replay, because the failure of checkpoint. " +
```
[the remark this answers is not in the document]
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java]: I changed the mtime to create time.

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java (hunk 4 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1234]: If free disk size is less than 5G, FE will not start.

**Review by nshangyiming — APPROVED** [review]

**Comment by gengjun-git** [comment]
@megifyio rebase

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**